test/schema_statements/changing_column_type_test.rb

```diff
@@ -260,6 +260,22 @@ def test_finalize_column_type_change_copies_check_constraints
       end
     end
 
+    def test_finalize_column_type_change_preserves_not_null_without_default_before_12
+      with_postgres(11) do
+        @connection.initialize_column_type_change(:projects, :description, :text)
+        @connection.finalize_column_type_change(:projects, :description)
+        refute column_for(:projects, :description).null
+      end
+    end
+
+    def test_finalize_column_type_change_preserves_not_null_without_default_after_12
```

Comment:
I couldn't figure out how to verify that >= 12 doesn't update `pg_attribute`...

Tried to explicitly `revoke update on pg_catalog.pg_attribute on #{@connection.raw_connection.user`, but this didn't seem to take.

As such, the `_before_12` and `_after_12` specs are identical in this first draft, except for the version specification.